### ISSUE TITLE
Compatibility with pystac 1.8

### ISSUE THREAD

**Issue by TomAugspurger** [stac-utils/pystac-client#562]
CI is currently failing with a mypy check

```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

pystac_client/collection_client.py:117: error: Signature of "get_items" incompatible with supertype "Catalog"  [override]
pystac_client/collection_client.py:117: note:      Superclass:
pystac_client/collection_client.py:117: note:          def get_items(self, *ids: str, recursive: bool = ...) -> Iterator[Item]
pystac_client/collection_client.py:117: note:      Subclass:
pystac_client/collection_client.py:117: note:          def get_items(self) -> Iterator[Item]
pystac_client/client.py:4[42](https://github.com/stac-utils/pystac-client/actions/runs/5570293343/jobs/10174236182?pr=561#step:5:43): error: Signature of "get_items" incompatible with supertype "Catalog"  [override]
pystac_client/client.py:[44](https://github.com/stac-utils/pystac-client/actions/runs/5570293343/jobs/10174236182?pr=561#step:5:45)2: note:      Superclass:
pystac_client/client.py:442: note:          def get_items(self, *ids: str, recursive: bool = ...) -> Iterator[Item]
pystac_client/client.py:442: note:      Subclass:
pystac_client/client.py:442: note:          def get_items(self) -> Iterator[Item]
Found 2 errors in 2 files (checked 21 source files)
```

@gadomski do you have a preference on how many hurdles to jump through to support this? With some effort, we could support older versions of pystac (using a mix of `*args`, `**kwargs`, and runtime version checks). Or we could just bump the required version of pystac.

My preference is to just bump the required version of pystac.

**Comment by jsignell** [comment]
Pete is out for a week or so, but I am :+1: on bumping the required version of pystac.

**Comment by TomAugspurger** [comment]
Thanks, I'll take a look at that PR too.

BTW, I hope SciPy was fun! I was bummed I couldn't make it.

**Comment by TomAugspurger** [comment]
Closed by #559
